Classes/Traits/InjectIconFactoryTrait.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/jwtools2.
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\Jwtools2\Traits;
+
+use TYPO3\CMS\Core\Imaging\IconFactory;
+
+/**
+ * Trait to inject IconFactory
+ */
+trait InjectIconFactoryTrait
+{
+    protected ?IconFactory $iconFactory = null;
```

Comment:
nullable? Why. It's part of the initialization process?

Reply:
![image](https://github.com/jweiland-net/jwtools2/assets/144037456/7f88a710-95b5-4e09-a543-91c554bb3a6c)

https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/DependencyInjection/Index.html#method-injection